### PR TITLE
Add tracing support

### PR DESCRIPTION
**Pull request by joe-elliott**
Adds support for pushing traces to a an OTel compatible backend.

- Adds configuration/tests
- Vendors Otel Collector and makes use of its pipeline configuration
- Adds a docker-compose example and a sample dashboard

### CODE REVIEW

File: pkg/tempo/tempo.go
Comment (rfratto) [inline pkg/tempo/tempo.go:154]: Can we do something so there's always a `component=tempo` KVP in the logs from here?

**Review by rfratto — COMMENTED** [review]
This is great, thank you for working on this! This is just a first pass; give me extra time to look into the config generation tomorrow. I might want to reduce the surface area we expose for defining the collectors.

File: docs/configuration-reference.md
Comment (joe-elliott) [reply docs/configuration-reference.md:1457]: It's GRPC so I don't think you can specify TLS through the scheme.

Comment (rfratto) [reply docs/configuration-reference.md:1457]: You're right, but we can parse it before passing it along to OTel.

Comment (rfratto) [reply docs/configuration-reference.md:1457]: Joe and I chatted offline and agreed that `url` with a scheme is confusing because it might appear like this is using HTTP/1 instead of gRPC (which doesn't support schemes). We agreed to go for a combo of `endpoint`/`insecure`

File: pkg/tempo/config.go
Comment (rfratto) [inline pkg/tempo/config.go:36]: Can you import this from [prometheus/common](https://github.com/prometheus/common/blob/f39dfa2b000545b3a763b844012958c275a62266/config/http_config.go#L41) instead?

File: pkg/tempo/config.go
Comment (rfratto) [inline pkg/tempo/config.go:81]:
```suggestion
				return nil, fmt.Errorf("unable to load password file %s: %w", c.RemoteWrite.BasicAuth.PasswordFile, err)
```

File: pkg/tempo/config.go
Comment (rfratto) [inline pkg/tempo/config.go:138]:
```suggestion
		return nil, fmt.Errorf("failed to merge in mapstructure config: %w", err)
```

File: pkg/tempo/config.go
Comment (rfratto) [inline pkg/tempo/config.go:143]:
```suggestion
		return nil, fmt.Errorf("failed to create factories: %w", err)
```

File: pkg/tempo/config.go
Comment (rfratto) [inline pkg/tempo/config.go:148]:
```suggestion
		return nil, fmt.Errorf("failed to load OTel config: %w", err)
```

File: pkg/tempo/config_test.go
Comment (rfratto) [inline pkg/tempo/config_test.go:12]: Can this use `yaml.v2` instead so we don't have to vendor two different versions?

**Review by rfratto — COMMENTED** [review]
Final pass! This is looking good; I mostly have nits but we should be able to merge this soon.

File: pkg/tempo/tempo.go
Comment (joe-elliott) [reply pkg/tempo/tempo.go:24]: yes :)

File: pkg/tempo/config_test.go
Comment (joe-elliott) [reply pkg/tempo/config_test.go:196]: Will update but to confirm I'm not crazy I got this pattern from the uber style guide: https://github.com/uber-go/guide/blob/master/style.md#test-tables

`tc` and `tt` do make more sense though.

File: docs/configuration-reference.md
Comment (joe-elliott) [reply docs/configuration-reference.md]: batch and queue aren't bad.  copying out the entire receivers config would be a lot of content.  I feel like that would clutter.

Also the goal would be to update to otel 0.9.0+ as soon as possible.  (As soon as etcd updates its GRPC dependency) and then we'd have to rewrite it.

File: go.mod
Comment (joe-elliott) [reply go.mod]: seems like that accidentally snuck in for a test.  removed.

File: docs/configuration-reference.md
Comment (rfratto) [reply docs/configuration-reference.md]: Alright, let's leave this for now as-is 👍

File: docs/configuration-reference.md
Comment (rfratto) [inline docs/configuration-reference.md:1454]:
```suggestion
  endpoint: <string>
```

File: docs/getting-started.md
Comment (rfratto) [inline docs/getting-started.md:247]:
```suggestion
    endpoint: localhost:55680
```

File: docs/overview.md
Comment (rfratto) [inline docs/overview.md:80]: Nice, thanks. Can we also note which non-Tempo systems the Agent can write spans to?

**Review by rfratto — APPROVED** [review]
LGTM minus remaining comments. I'm planning a 0.6.1 patch release tomorrow and I'd like to hold off on merging this until after that's out.

**Comment by rfratto** [comment]
@joe-elliott before I merge, can you add credit to yourself in the CHANGELOG?

**Comment by joe-elliott** [comment]
@rfratto added a note in the changelog

**Pull request merged**
